lightly/api/api_workflow_upload_embeddings.py

```diff
@@ -67,11 +68,21 @@ def upload_embeddings(self, path_to_embeddings_csv: str, name: str):
         self.embedding_id = response.embedding_id
         signed_write_url = response.signed_write_url
 
-        # upload the csv to the URL
-        with open(path_to_ordered_embeddings_csv, 'rb') as file_ordered_embeddings_csv:
-            self.upload_file_with_signed_url(file=file_ordered_embeddings_csv, signed_write_url=signed_write_url)
+        # save the csv rows in a temporary in-memory string file
+        # using a csv writer and then read them as bytes
+        with tempfile.SpooledTemporaryFile(mode="rw") as f:
+            writer = csv.writer(f)
+            writer.writerows(rows_csv)
+            f.seek(0)
+            embeddings_csv_as_bytes = f.read().encode('utf-8')
```

Comment:
why can we not directly use this file for the upload?

Reply:
I tried, but we need to write in a string file, but need a byte file for upload. The TemporarySpoolFile (ans similar, BytesIO, StringIO) don't allow mixing strings and bytes.

Reply:
Anyway, as you see from the performance tree diagram, not this conversion (reading from StringFile, converting to bytes, writing to BytesFile) is taking long, but only the actual write with the csv writer. It might be further optimised by creating the strings for each line without the csv writer, but I don't think we need further peformance improvements at the moment.

Reply:
Ok, we can still check alternatives when this becomes a bottleneck